contracts/pool/src/contract.rs

```diff
@@ -883,10 +887,53 @@ impl LiquidityPool {
 
     #[allow(dead_code)]
     //TODO: Remove after we've added the key to storage
-    pub fn add_new_key_to_storage(env: Env) -> Result<(), ContractError> {
+    pub fn add_contract_name_key_to_storage(env: Env) -> Result<(), ContractError> {
         env.storage().persistent().set(&XYK_POOL_KEY, &true);
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub fn extend_all_tll(env: Env) -> Result<(), ContractError> {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_RENEWAL_THRESHOLD, INSTANCE_TARGET_TTL);
+
+        env.storage().persistent().extend_ttl(
+            &DEFAULT_SLIPPAGE_BPS,
+            PERSISTENT_RENEWAL_THRESHOLD,
+            PERSISTENT_TARGET_TTL,
+        );
+
+        env.storage().persistent().extend_ttl(
+            &CONFIG,
+            PERSISTENT_RENEWAL_THRESHOLD,
+            PERSISTENT_TARGET_TTL,
+        );
+
+        for val in &[
+            DataKey::TotalShares,
+            DataKey::ReserveA,
+            DataKey::ReserveB,
+            DataKey::Admin,
+            DataKey::Initialized,
+        ] {
+            env.storage().persistent().extend_ttl(
+                val,
+                PERSISTENT_RENEWAL_THRESHOLD,
+                PERSISTENT_TARGET_TTL,
+            )
+        }
```

Comment:
And here for comparison you are not checking for the key existence as well.
I think you should remove that error from the factory.